core/src/com/unciv/logic/automation/unit/BattleHelper.kt

```diff
@@ -121,8 +121,8 @@ object BattleHelper {
         if (tileCombatant is CityCombatant && tileCombatant.city.hasJustBeenConquered) return false
         if (!combatant.getCivInfo().isAtWarWith(tileCombatant.getCivInfo())) return false
 
-        if (combatant is MapUnitCombatant && combatant.isLandUnit() && combatant.isMelee() &&
-            !combatant.hasUnique(UniqueType.LandUnitEmbarkation) && tile.isWater
+        if (combatant is MapUnitCombatant && combatant.isLandUnit() && combatant.isMelee() && tile.isWater &&
+            (!combatant.getCivInfo().tech.unitsCanEmbark && !combatant.unit.cache.canMoveOnWater)
```

Comment:
Unnecessary extra parentheses
We can separate these into different lines for clarity

Reply:
I'm not sure much is gauned from splitting to another line. First line is situation (melee land unit vs water tile), second is disqualifier (can't embark here and can't naturally move on water)